feat: uniform layout and naming of hints, remarks, exceptions (#660) (#860)
zalando/restful-api-guidelines: chapters/events.adoc

<version>               ::= [Vv][0-9.]* -- major version of non compatible schemas
----

> TIP: The following convention (e.g. used by legacy STUPS infrastructure) is deprecated

Reviewer: ```suggestion
*Historic information*: The following convention (e.g. used by legacy STUPS infrastructure) is deprecated
```

information (the contents of <<event-metadata,metadata>> are
described further down in this section).

> [NOTE]

Reviewer: Let's revert this change.

 * a strictly monotonically increasing sequence counter (maintained per partition
   or event type).

TIP: Timestamps are often a bad choice, since in distributed systems events
may occur at the same time, or clocks are not exactly synchronized, or
jump forward and backward to compensate drifts or leap-seconds. If you use anyway
timestamps to indicate event ordering, you _must_ carefully ensure that the designated
event order is not messed up by these effects and use UTC time zone format.

> NOTE: The `received_at` and `partition_offset` metadata set by Nakadi typically is

Reviewer: ```suggestion
The `received_at` and `partition_offset` metadata set by Nakadi typically is
```

customer behavior reporting (click rates) might not care about event duplicates 
since accuracy in per-mille range is not needed.

[TIP]

> ====

Reviewer: ```suggestion
```

Event consumers are supported in deduplication:

* Deduplication can be based on the `eid` as mandatory standard for all events -- see <<211>>.
* Processing data change events to replay data changes and keep transactional
data copies in sync (CDC) is robust against duplicates because it is based on
data keys and change ordering -- see <<202>> and <<242>>.
* Data analytics users of the Data Lake are well advised to use curated data as a
source for analytics. Raw event datasets materialized in the lake are typically
cleaned-up (including deduplication and data synchronization) and provided as
transactional data copy or curated data products, for instance
https://curated-product-data.docs.zalando.net/[curated product data [internal link]] or
https://curated-sales-data.docs.zalando.net/[curated sales data [internal link]].

> ====

Reviewer: ```suggestion
```
